internal/controller/mandatorymodule/deletion_controller.go

```diff
@@ -20,152 +20,37 @@ import (
-	}
-
-	if template.DeletionTimestamp.IsZero() {
-		return ctrl.Result{}, nil
-	}
-
-	mrm, err := r.GetModuleReleaseMeta(ctx, template.Spec.ModuleName, template.Namespace)
-	if err != nil {
-		return ctrl.Result{}, fmt.Errorf("failed to find ModuleReleaseMeta for Mandatory Module %s: %w",
-			template.Name, err)
-	}
-	ocmId, err := ocmidentity.NewComponentId(mrm.Spec.OcmComponentName, template.Spec.Version)
-	if err != nil {
-		return ctrl.Result{}, fmt.Errorf("failed to create OCM identity for Mandatory Module %s: %w",
-			template.Spec.ModuleName, err)
-	}
-
-	manifests, err := r.getCorrespondingManifests(ctx, template.Namespace, *ocmId)
-	if err != nil {
-		return ctrl.Result{}, fmt.Errorf("failed to get MandatoryModuleManifests: %w", err)
-	}
-
-	if len(manifests) == 0 {
-		if controllerutil.RemoveFinalizer(template, shared.MandatoryModuleFinalizer) {
-			return r.updateTemplateFinalizer(ctx, template)
-		}
-		return ctrl.Result{}, nil
-	}
-
-	if err := r.removeManifests(ctx, manifests); err != nil {
-		r.Event.Warning(template, deletingManifestError, err)
-		return ctrl.Result{}, fmt.Errorf("failed to remove MandatoryModule Manifest: %w", err)
-	}
-
-	return ctrl.Result{Requeue: true}, nil
-}
-
-func (r *DeletionReconciler) GetModuleReleaseMeta(ctx context.Context, moduleName, namespace string) (
-	*v1beta2.ModuleReleaseMeta, error,
-) {
-	return templatelookup.GetModuleReleaseMeta(ctx, r.Client, moduleName, namespace)
+type DeletionReconciler struct {
+	deletionService  DeletionService
+	requeueIntervals queue.RequeueIntervals
 }
 
-func (r *DeletionReconciler) updateTemplateFinalizer(ctx context.Context,
-	template *v1beta2.ModuleTemplate,
-) (ctrl.Result, error) {
-	if err := r.Update(ctx, template); err != nil {
-		r.Event.Warning(template, settingFinalizerError, err)
-		return ctrl.Result{}, fmt.Errorf("failed to update MandatoryModuleTemplate finalizer: %w", err)
+func NewDeletionReconciler(deletionService DeletionService,
+	requeueIntervals queue.RequeueIntervals,
+) *DeletionReconciler {
+	return &DeletionReconciler{
+		deletionService:  deletionService,
+		requeueIntervals: requeueIntervals,
 	}
-	return ctrl.Result{Requeue: true}, nil
 }
 
-func (r *DeletionReconciler) getCorrespondingManifests(ctx context.Context,
-	namespace string, ocmId ocmidentity.ComponentId) ([]v1beta2.Manifest,
-	error,
-) {
-	manifests := &v1beta2.ManifestList{}
-	if err := r.List(ctx, manifests, &client.ListOptions{
-		Namespace:     namespace,
-		LabelSelector: k8slabels.SelectorFromSet(k8slabels.Set{shared.IsMandatoryModule: "true"}),
-	}); client.IgnoreNotFound(err) != nil {
-		return nil, fmt.Errorf("not able to list mandatory module manifests: %w", err)
-	}
-
-	filtered := filterManifestsByComponentIdentity(manifests.Items, ocmId)
-
-	return filtered, nil
+func (r *DeletionReconciler) Reconcile(ctx context.Context, mrm *v1beta2.ModuleReleaseMeta) (ctrl.Result, error) {
+	err := r.deletionService.HandleDeletion(ctx, mrm)
```

Comment:
guard statement here? maybe even panic since it's a developer error that MRM is in deletion?